src/Types.mo

```diff
@@ -61,7 +61,7 @@ module {
       children : [var ?Node<K, V>]
     };
 
-    public type Leaf<K, V> = {
+    public type Leaf<K, V> = { // why the extra indirection?
```

Comment:
Seems like we could remove `Leaf<K, V>` and then rename `Data` to `Leaf` for clarity?

Reply:
Yeah, but perhaps Luc was intending to keep compatibility with the original library. If we do this, I would keep the Data name and just remove the Leaf type (since internal nodes also carry Data).

Reply:
@luc-blaeser are we free to simplify the representation or do we want to preserve compatibility with Byron's representation for easier migration?